hoodie-spark/src/main/scala/com/uber/hoodie/IncrementalRelation.scala

```diff
@@ -92,12 +96,15 @@ class IncrementalRelation(val sqlContext: SQLContext,
     // will filter out all the files incorrectly.
     sqlContext.sparkContext.hadoopConfiguration.unset("mapreduce.input.pathFilter.class");
     val sOpts = optParams.filter(p => !p._1.equalsIgnoreCase("path"))
-    sqlContext.read.options(sOpts)
-      .schema(latestSchema) // avoid AnalysisException for empty input
-      .parquet(fileIdToFullPath.values.toList: _*)
-      .filter(String.format("%s >= '%s'", HoodieRecord.COMMIT_TIME_METADATA_FIELD, commitsToReturn.head.getTimestamp))
-      .filter(String.format("%s <= '%s'", HoodieRecord.COMMIT_TIME_METADATA_FIELD, commitsToReturn.last.getTimestamp))
-      .toDF().rdd
-
+    if (fileIdToFullPath.isEmpty) {
+      sqlContext.sparkContext.emptyRDD[Row]
+    } else {
+      sqlContext.read.options(sOpts)
+        .schema(latestSchema) // avoid AnalysisException for empty input
```

Comment:
Could you explain that comment please ? Not sure how this is avoiding that exception..

Reply:
Ohh, the comment is probably more suited in the if block ?

Reply:
Done